### ISSUE TITLE
Implement a function to send messages to the user

### ISSUE THREAD

**Issue by Abuelodelanada** [salesagility/SuiteCRM#3048]
#### Expected Behavior
Sometimes is usefull to send messages to the user when something happens.
Drupal has a function called drupal_set_message() for this:

https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/drupal_set_message/7.x

We have implemented this functionality in a SugarCRM CE instance, and use it to warn the user, or to show the user an error happend, or just to send information.

Here are an example showing the 5 types of messages we have implemented
![suite_set_message](https://cloud.githubusercontent.com/assets/939888/22603830/bfbb0b98-ea27-11e6-9572-44121fd6d880.png)

This feature is manly used in controller.php and views. Here are a real life example:

![sugar_set_message_function](https://cloud.githubusercontent.com/assets/939888/22604323/c3eba428-ea29-11e6-8db1-3188d8d1743f.png)

**Comment by pgorod** [comment]
Looks good! Is that code something that you could contribute to the project?

There are several cases of weird messages in SuiteCRM. Sometimes with errors we get blank screens, sometimes a text message, sometimes a pop-up,; or when people click on a link in an email and they get a screen with just a phrase saying that "a meeting has been accepted" or something. Then there is some more generic information flowing on the activity dashlet.

It would be great to get everything more coherent and better looking.

Another option that could be considered is injecting a message into that little bubble on the top menu (notifications). Less visible than your solution, but perhaps more integrated into what is already used now.

**Comment by Abuelodelanada** [comment]
Yes, we have to clean and order the code before contributing...but I create the issue to reference it from the pull request.

This was thinked initialy for errors in saving...many times before saving or doing an action we have to make a request to an external Web Servise...so it is usefull to show information

**Comment by Abuelodelanada** [comment]
I think the notifications in top menu is for other kind of information, for example when someone assing you a record or something like that

**Comment by Abuelodelanada** [comment]
@pgorod take a look at PR #3078

**Comment by pgorod** [comment]
Hey thanks, that looks great!

We'll have to start thinking when we should call these functions. What I mean is, what kinds of messages currently in the code would be better displayed using this method. This would be in a different PR, I believe, but maybe we can get the discussion going here.

**Comment by Abuelodelanada** [comment]
@pgorod we use it mostly for validations when saving a record, to show errors when make a request to an external web service, or to show information about the record like "This record is inmutable, you can't edit it" or....

I think we can use it to replace the red messages sometimes SuiteCRM shows when something is bad...

**Comment by marin-h** [comment]
> I think we can use it to replace the red messages sometimes SuiteCRM shows when something is 
> bad...

I agree with @Abuelodelanada, actually the implementation of displayAdminError method in modules/Administration/DisplayWarnings.php isn't a lot different from the one used in #3078... it would be easy I think to adapt that to use these new functions. 

I'm attaching an example of displayAdminError below, showing a message using the same CSS styling as the 'error' class created for #3078 . 
![captura de pantalla_2017-02-10_14-59-24](https://cloud.githubusercontent.com/assets/5542756/22837951/8fec8bce-efa1-11e6-8a8c-1920fd55d773.png)

**Comment by pgorod** [comment]
For me, two instances that really need improvement when communicating with users are these:

1 - When I **click on the link sent in emails regarding calls or meetings**, I get a blank white screen with nothing but this:

![accepted](https://cloud.githubusercontent.com/assets/15945027/22838592/c5ec0fe4-efbd-11e6-8c57-a7c374535aba.jpg)

This is a special case because there might be no user logged in. But if there is, it would be better to show some UI, menus, and that new message bar.

2 - **Quick Repair and Rebuild**. The message bar should appear across the top when there are warnings or error messages. So often there is an important error dozens of lines down, forcing you to scroll every time. There should be a final message across the top saying the aggregate result of all the actions taken (green/yellow/red according for success/warnings/actions required).

And here's an additional number 3 - maybe **the pop-ups** could also be changed to banners, at least when they are informational and only have an "Ok" button to acknowledge. The Yes/No and the Ok/Cancel pop-ups are more complicated. They could at least be improved visually, and maybe even with graphical similarity to these bars.

These are just ideas...

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Implement a function to send messages to the user](https://trello.com/c/Rcjss5a6/33-implement-a-function-to-send-messages-to-the-user)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
